ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Client.scala

```diff
@@ -313,6 +313,27 @@ private[ember] class H2Client[F[_]](
 
 private[ember] object H2Client {
   private type TinyClient[F[_]] = Request[F] => Resource[F, Response[F]]
+
+  private type ConnectionPool[F[_]] = Map[H2Client.RequestKey, (H2Connection[F], F[Unit])]
```

Comment:
It looks too specialized and won't be reused, so I would make it private with the most restricted scope.

Reply:
Written into the function.